### DISCUSSION TITLE
General DiscuFrequent interactions between each node of workflow and DB impacting performancession

### DISCUSSION THREAD

**Discussion by YidaHu** [langgenius/dify#9668] [General]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Content

I have observed that when a large number of nodes are configured in a workflow, the frequent interactions with the database significantly increase the time consumption.

As the workflow progresses with many nodes, the constant back-and-forth with the database becomes a bottleneck. For example, [mention specific scenarios where the increased time due to DB interactions is evident, such as a long-running process that could be sped up if DB interactions were reduced].

Is it possible to minimize such DB interaction operations? It would be ideal if the tracking data could be written to the DB at the completion of execution instead of having interactions at every node execution.

This approach would likely improve the overall efficiency and speed of the workflow, reducing unnecessary delays caused by excessive DB access.

**Comment by crazywoola** [comment]
@takatost

**Reply by YidaHu** [reply]
> @takatost

@takatost Hi.  Is there a plan to deal with this issue?
